lib/OptimizationBBO/src/OptimizationBBO.jl

```diff
@@ -44,16 +46,52 @@ function __map_optimizer_args(prob::SciMLBase.OptimizationProblem, opt::BBO;
         mapped_args = (; mapped_args..., MinDeltaFitnessTolerance=abstol)
     end
 
+    if !isnothing(verbose)
+        mapped_args = (; mapped_args..., TraceMode=verbose)
+    end
+
     return mapped_args
 end
 
+function log_lines(rd, progress, maxiters, id)
+    while isopen(rd)
+        ln = readline(rd)
+        isempty(ln) && continue
+        m = match(r"([0-9]*) steps", ln)
+        if !isnothing(m)
+            s = tryparse(Int, m[1])
+            isnothing(s) && continue
+            isnothing(maxiters) && continue
+            progress && @info ln progress=(s / maxiters) _id=id
+        end
+    end
+end
+
+function redirect_output(f::Function, progress, maxiters, id)
+    old_stdout = stdout
+    rd, = redirect_stdout()
+    task = @async log_lines(rd, progress, maxiters, id)
+    try
+        ret = f()
+        Libc.flush_cstdio()
+        flush(stdout)
+        return ret
+    finally
+        close(rd)
+        redirect_stdout(old_stdout)
+        wait(task)
+    end
+end
+
 function SciMLBase.__solve(prob::SciMLBase.OptimizationProblem, opt::BBO, data=Optimization.DEFAULT_DATA;
     callback=(args...) -> (false),
     maxiters::Union{Number,Nothing}=nothing,
     maxtime::Union{Number,Nothing}=nothing,
     abstol::Union{Number,Nothing}=nothing,
     reltol::Union{Number,Nothing}=nothing,
-    progress=false, kwargs...)
+    verbose::Symbol=:compact,
```

Comment:
we only allow true/false

Reply:
Do we separate between `progress` and `verbose`?

I'm not sure how to handle 
- `progress=false`, `verbose=true`. Do we show all logs with `@info` here?
- `progress=true`, `verbose=false`. Do we just show the progress here?
- `progress=true`, `verbose=true`. And here we also add other info like in the screenshot?

Reply:
Progress and verbose are different. Trace logs should all be handled by progress. Verbose should be left to things like throwing warnings.

Reply:
Ok, so in that case all the functionality of the PR should be captured by `progress` and `verbose` will be reserved for warnings?
For the progress, should we display anything besides number of steps/maxiters?

Reply:
@ChrisRackauckas Should I remove `verbose` from this PR and just always redirect the `stdout` and use it for progress logging when `progress=true`?

Reply:
We shouldn't just redirect to stdout. We should always control it with progress and verbose ourselves.

Reply:
Thanks for the comment. I reimplemented the PR using the callback and querying the `BlackBoxOptim.OptRunController` for info. I've also implemented time based or maxiters based progress.